### PR TITLE
Changed DataLayer structs to classes

### PR DESCRIPTION
**Pull request by jtdkhamp**

### CODE REVIEW

File: src/DataLayer/BatteryData.h [hunk 1]
```diff
@@ -1,91 +1,136 @@
...
-struct BatteryData
+#include <QObject>
+
+enum PrechargeState
+{
+    Idle = 0,
+    Precharge = 1,
+    Measure = 2,
+    EnablePack = 3,
+    Run = 4,
```
Comment (Reddragoneer1) [inline src/DataLayer/BatteryData.h]: Is it fine that there's a comma here?

File: src/DataLayer/KeyMotorData.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    , motor0BusCurrent_(3.3f)
+    , motor0BusVoltage_(4.4f)
+    , motor0VehicleVelocity_(5.5f)
+    , motor1Alive_(false)
+    , motor1SetCurrent_(6.6f)
+    , motor1SetVelocity_(7.7f)
+    , motor1BusCurrent_(8.8f)
+    , motor1BusVoltage_(9.9f)
+    , motor1VehicleVelocity_(10.01f)
+{
```
Comment (Reddragoneer1) [inline src/DataLayer/KeyMotorData.cpp]: Can someone tell me why the above code isn't in the bracket? It looks as though it should be

Comment (bill-luu) [reply src/DataLayer/KeyMotorData.cpp]: It's an initialization list

**Comment by bill-luu** [comment]
Continued in #49

**Pull request closed without merging**